src/agent0/ethpy/base/transactions.py

```diff
@@ -419,21 +424,30 @@ def build_transaction(
 
     # This is the additional transaction argument passed into function.call
     # that may contain additional call arguments such as max_gas, nonce, etc.
+    transaction_kwargs = TxParams(
+        {
+            "from": signer_checksum_address,
+            "nonce": nonce,
+        }
+    )
     if txn_options_value is not None:
-        transaction_kwargs = TxParams(
-            {
-                "from": signer_checksum_address,
-                "nonce": nonce,
-                "value": Wei(txn_options_value),
-            }
-        )
+        transaction_kwargs["value"] = Wei(txn_options_value)
+
+    # Assign gas parameters
+    max_priority_fee = web3.eth.max_priority_fee
+    pending_block = web3.eth.get_block("pending")
+    base_fee = pending_block.get("baseFeePerGas", None)
+    if base_fee is None:
+        raise AssertionError("The latest block does not have a baseFeePerGas")
+    max_fee_per_gas = max_priority_fee + base_fee
+
+    if txn_options_gas is None:
+        transaction_kwargs["gas"] = int(web3.eth.estimate_gas(transaction_kwargs) * GAS_ESTIMATE_MULTIPLIER)
     else:
-        transaction_kwargs = TxParams(
-            {
-                "from": signer_checksum_address,
-                "nonce": nonce,
-            }
-        )
+        transaction_kwargs["gas"] = txn_options_gas
+
+    transaction_kwargs["maxFeePerGas"] = Wei(max_fee_per_gas)
```

Comment:
Did we determine if these fields are needed?

Reply:
no they're not, they're just being more explicit about default behavior. Pretty sure I took them from somewhere else in our code.

Reply:
added clarification that we're exposing default web3py behavior (other than the gas parameter) in 6c05738bd101660aa9d51c745e3837906f098736

Reply:
deleted the estimate_gas portion here because it breaks on devnet: 39580eb0cd3c07282add2957a6e5258cd4670683